models/evc.py

```diff
@@ -1320,33 +1422,69 @@ def check_trace(circuit, trace_a, trace_z):
         return True
 
     @staticmethod
-    def check_list_traces(list_circuits):
+    def check_range(circuit, traces: list) -> bool:
+        """Check traces when for UNI with TAGRange"""
+        check = True
+        mask_list = (circuit.uni_a.user_tag.mask_list or
+                     circuit.uni_z.user_tag.mask_list)
+        for i, mask in enumerate(mask_list):
+            trace_a = traces[i*2]
+            trace_z = traces[i*2+1]
+            check &= EVCDeploy.check_trace(
+                mask, mask,
+                circuit.uni_a.interface,
+                circuit.uni_z.interface,
+                circuit.current_path,
+                trace_a, trace_z,
+            )
+        return check
+
+    @staticmethod
+    def check_list_traces(list_circuits: list) -> dict:
         """Check if current_path is deployed comparing with SDN traces."""
         if not list_circuits:
             return {}
-        uni_list = []
-        for circuit in list_circuits:
-            uni_list.append(circuit.uni_a)
-            uni_list.append(circuit.uni_z)
-
-        traces = EVCDeploy.run_bulk_sdntraces(uni_list)
-        traces = traces["result"]
-        circuits_checked = {}
+        uni_list = make_uni_list(list_circuits)
+        traces = EVCDeploy.run_bulk_sdntraces(uni_list)["result"]
+
         if not traces:
-            return circuits_checked
+            return {}
 
         try:
-            for i, circuit in enumerate(list_circuits):
-                trace_a = traces[2*i]
-                trace_z = traces[2*i+1]
-                circuits_checked[circuit.id] = EVCDeploy.check_trace(
-                        circuit, trace_a, trace_z
+            circuits_checked = {}
+            i = 0
+            for circuit in list_circuits:
+                if isinstance(circuit.uni_a.user_tag, TAGRange):
+                    length = (len(circuit.uni_a.user_tag.mask_list) or
```

Comment:
Up to this point don't we have validation guarantees during EVC creation and updates that all EVCs will have symmetric mask_list? If this requirement ever change I can see this becoming difficult to remember and identify, even though in some cases it'll result in index error. Let me know if you're seeing any other case or had any other reason to compute the length with this `or`. If you want extra guarantees here, maybe if the lengths are diff then you raise an exception just so it's immediately clear that's not supported?

Reply:
Added this because I was creating TAGs in kytos console with empty `mask_list`s to test exceptions. Currently without any modification to an EVC, both `value` and `mask_list` should be the same so this `len() or len()` is not should not be needed. Going to delete it.

Reply:
Got it. OK.

Reply:
Updated, commit [918e21324fdd2dc3d6615cf967613ae75802fb31](https://github.com/kytos-ng/mef_eline/pull/407/commits/918e21324fdd2dc3d6615cf967613ae75802fb31)

Reply:
Made a mistake with length, fixed commit [1613b6f110a123dca424d374c8d7ea735be13bb3](https://github.com/kytos-ng/mef_eline/pull/407/commits/1613b6f110a123dca424d374c8d7ea735be13bb3)